src/Compilers/CSharp/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -1038,7 +1038,8 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                 languageVersion: languageVersion,
                 preprocessorSymbols: defines.ToImmutableAndFree(),
                 documentationMode: parseDocumentationComments ? DocumentationMode.Diagnose : DocumentationMode.None,
-                kind: SourceCodeKind.Regular
+                kind: SourceCodeKind.Regular,
+                features: features.ToImmutableDictionary(feature => feature, feature => "true")
```

Comment:
The command line parser currently doesn't do much /feature parameter parsing. Expected behavior is specified in a test, this assert in particular is interesting.

https://github.com/dotnet/roslyn/blob/80d2e0bf26ef5eddea43b1e9b506e8b604bf5db5/src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs#L6856

I think @jaredpar might have more to say on this (with respect to having an ImmutableArray in CompilationOptions and Dictionary in ParseOptions)